fe/fe-core/src/main/java/com/starrocks/mysql/privilege/AuthPlugin.java

```diff
@@ -18,5 +18,6 @@
 public enum AuthPlugin {
     MYSQL_NATIVE_PASSWORD,
     AUTHENTICATION_LDAP_SIMPLE,
-    AUTHENTICATION_KERBEROS
+    AUTHENTICATION_KERBEROS,
+    AUTHENTICATION_LDAP_SIMPLE_FOR_EXTERNAL
```

Comment:
Can we use a more appropriate name? I don't think the two auth should be distinguished by internal/external. Instead, functions should be used to differentiate. Such as AUTHENTICATION_LDAP_MAPPING or AUTHENTICATION_LDAP_EXTERNAL

Reply:
the reason that why we use a new AuthenticationProvider is that we gonna to extract some core logic from AUTHENTICATION_LDAP_SIMPLE_FOR_EXTERNAL and put it into the enterprise edition, for compatibility, we have to use a new provider without affecting the community users, and i don't want to mixup the new feature with the old provider, because in the long run, we may remove the old provider in the future.

Reply:
about the naming, i think it's ok, and it "FOR_NATIVE" not "FOR_INTERNAL", "EXTERNAL" just means that the user metadata is stored on external server, i don't want to use a too specific name like "FOR_LDAP_MAPPING", because maybe in the future, we don't use it to do group mapping and just use the provider to authenticate.